### PR TITLE
Fix handling of scoped eslint plugins

### PR DESCRIPTION
**Pull request by corbinu**
I ran into some issues of depcheck showing my scoped plugin which contains my personal eslint config as both being unused and missing dependancies. 

It appears to be related to that the config is stored in a scoped plugin. This is the fix.

### CODE REVIEW

File: src/special/eslint.js [hunk 1]
```diff
@@ -61,6 +65,15 @@ function resolvePresetPackage(preset, rootDir) {
   if (isEslintConfigARelativePath(preset)) {
```
Comment (corbinu) [inline src/special/eslint.js]: Sure will do

**Comment by lijunle** [comment]
@corbinu 

Before you start fixing the comments. Could you please consider this suggestion? It is more important than fixing any comment!

> Thank you for contributing this project. I looked into [the current ESLint resolve logic](https://github.com/eslint/eslint/blob/da79f17bf24de631bb24ef422c8dada7261645c4/lib/config/config-file.js#L461). I find it is too complicated to shadow it here.
> My suggestion is to reference the ESLint package and logic directly. How do you think about this?

**Comment by corbinu** [comment]
@lijunle I will take a look but wanted to make the other changes first ... as once that is done we can try simply replacing resolvePresetPackage(preset, rootDir) and see if it works. If it still passes then is great. Otherwise can move forward.

**Comment by lijunle** [comment]
@corbinu In such case, we could better to have some test cases. Here are the cases I identified from the current ESLint implementation:

```php
plugin:@my-scope				=> @my-scoped/eslint-plugin
plugin:@my-scope/eslint-plugin			=> @my-scoped/eslint-plugin
plugin:@my-scope/my-plugin			=> @my-scoped/eslint-plugin-my-plugin
plugin:@my-scope/eslint-pluginmy-plugin		=> @my-scoped/eslint-plugin-my-plugin
plugin:eslint-plugin-my-plugin			=> eslint-plugin-my-plugin
plugin:my-plugin				=> eslint-plugin-my-plugin
```

For `eslint-config`, it basically has the same logic without the `plugin:` prefix.

**Comment by corbinu** [comment]
Ok will take a look at the testing system when I have some time.

**Pull request closed without merging**